yearn/prices/curve.py

```diff
@@ -135,7 +135,9 @@ def watch_events(self):
                     self.registries[event.address].add(event['pool'])
                     lp_token = contract(event.address).get_lp_token(event['pool'])
                     self.token_to_pool[lp_token] = event['pool']
-
+                elif event.name == 'PoolRemoved':
+                    self.registries[event.address].remove(event['pool'])
+                    
```

Comment:
could you extract the `event['pool']` into a variable and check in L139 if it's in the list before removing ? Otherwise this might crash if it's not there for some reason.

Reply:
ah gotcha, didn't know that's how sets work in python

Reply:
Changed to `discard` as that looks like more what I intended from the documentation
```
Remove an element from a set if it is a member.

If the element is not a member, do nothing.
```